### PR TITLE
Remove * inserts to ease migration pains

### PR DESCRIPTION
**Pull request by rymurr**

### CODE REVIEW

File: bootstrap/005_table_migrations.sql [hunk 1]
```diff
@@ -84,3 +84,34 @@ BEGIN
...
+
+create or replace procedure internal.generate_insert_statement_cmd(target_schema varchar, target_table varchar, source_schema varchar, source_table varchar, where_clause varchar)
+returns string
+as
+$$
+begin
+  let columns string := (
+      SELECT LISTAGG('"' || COLUMN_NAME || '"', ', ')
+      FROM (
+      SELECT COLUMN_NAME, DATA_TYPE FROM INFORMATION_SCHEMA.COLUMNS WHERE TABLE_SCHEMA = :source_schema AND TABLE_NAME = :source_table
```
Comment (joshelser) [inline bootstrap/005_table_migrations.sql:97]: Do we want to force an `ORDER BY ORDINAL_POSITION` or is that already guaranteed reading from `INFORMATION_SCHEMA.COLUMNS`?

File: bootstrap/005_table_migrations.sql [hunk 2]
```diff
@@ -84,3 +84,34 @@ BEGIN
...
+create or replace procedure internal.generate_insert_statement(target_schema varchar, target_table varchar, source_schema varchar, source_table varchar, where_clause varchar)
+returns number
+as
+$$
+begin
+    let stmt varchar;
+    call internal.generate_insert_statement_cmd(:target_schema, :target_table, :source_schema, :source_table, :where_clause) into stmt;
+  execute immediate stmt;
+            let inserted number := (select * from TABLE(RESULT_SCAN(LAST_QUERY_ID())));
+  return :inserted;
```
Comment (joshelser) [inline bootstrap/005_table_migrations.sql:115]: nit: indentation

File: bootstrap/018_warehouse_updates.sql [hunk 3]
```diff
@@ -61,10 +61,12 @@ BEGIN
             newest_completed := (SELECT greatest(coalesce(max(SESSION_END), 0::TIMESTAMP), :newest_completed) FROM RAW_WH_EVT WHERE NOT INCOMPLETE);
             let run_id timestamp := (SELECT run_id FROM RAW_WH_EVT limit 1);
             TRUNCATE TABLE INTERNAL_REPORTING_MV.CLUSTER_AND_WAREHOUSE_SESSIONS_COMPLETE_AND_DAILY_INCOMPLETE;
-            insert into INTERNAL_REPORTING_MV.CLUSTER_AND_WAREHOUSE_SESSIONS_COMPLETE_AND_DAILY_INCOMPLETE SELECT * FROM RAW_WH_EVT WHERE INCOMPLETE OR SESSION_END = :newest_completed;
-            let new_INCOMPLETE number := (select * from TABLE(RESULT_SCAN(LAST_QUERY_ID())));
-            insert into INTERNAL_REPORTING_MV.CLUSTER_AND_WAREHOUSE_SESSIONS_COMPLETE_AND_DAILY select * from RAW_WH_EVT WHERE NOT INCOMPLETE AND SESSION_END <> :newest_completed;
-            let new_closed number := (select * from TABLE(RESULT_SCAN(LAST_QUERY_ID())));
+            let where_clause varchar := (select 'INCOMPLETE OR session_end = to_timestamp_ltz(\'' || :newest_completed || '\')');
```
Comment (joshelser) [inline bootstrap/018_warehouse_updates.sql:64]: Doesn't need to be a select query, does it? Couldn't it just be

```
let where_clause varchar := 'INCOMPLETE OR session_end = to_timestamp_ltz(\'' || :newest_completed || '\')';
```

**Review by joshelser — CHANGES_REQUESTED** [review]

File: bootstrap/005_table_migrations.sql (hunk 1 above)
Comment (rymurr) [reply bootstrap/005_table_migrations.sql:97]: no longer neccessary with new algorithm for inserts

File: bootstrap/005_table_migrations.sql (hunk 2 above)
Comment (rymurr) [reply bootstrap/005_table_migrations.sql:115]: fixed

File: bootstrap/018_warehouse_updates.sql (hunk 3 above)
Comment (rymurr) [reply bootstrap/018_warehouse_updates.sql:64]: possibly, the rules of snowsql confuse me. If it doesn't need the select then snowflake will optimise it away

Comment (joshelser) [reply bootstrap/018_warehouse_updates.sql:64]: fair point.

**Review by joshelser — APPROVED** [review]

**Pull request merged**